File: sections/14_core_patterns_mixed_contest/problems/p_common_deadline/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int n;
    long long limit;
    cin >> n >> limit;
    vector<long long> durations(n);
    for (long long &duration : durations) cin >> duration;
    sort(durations.begin(), durations.end());
    long long total = 0;
    int answer = 0;
    for (long long duration : durations) {
        if (total + duration > limit) break;
        total += duration;
        ++answer;
    }
    cout << answer << '\n';
}

File: sections/14_core_patterns_mixed_contest/problems/p_common_deadline/solution.py
import sys


def main() -> None:
    data = list(map(int, sys.stdin.buffer.read().split()))
    limit = data[1]
    total = answer = 0
    for duration in sorted(data[2:]):
        if total + duration > limit:
            break
        total += duration
        answer += 1
    print(answer)


if __name__ == "__main__":
    main()
